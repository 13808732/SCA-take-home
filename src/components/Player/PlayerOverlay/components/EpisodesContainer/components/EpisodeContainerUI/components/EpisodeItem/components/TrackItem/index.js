import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import styled from 'styled-components';
import spacing from 'styles/helpers/spacing';
import { Box, Flex } from 'components/Grid';
import Paragraph from 'components/Typography/Paragraph';
import screen from 'styles/helpers/media';
import { formatSecondsToTime } from 'utilities/helpers/dateTime';
import Header from 'components/Typography/Header';
import EpisodeTags from '../EpisodeTags';

function TrackItem({ item, imgName, background, podcastName, isEpisodeNew }) {
  const imageUrl = get(item, 'show.images.square.url', null) || item.imageUrl;
  return (
    <StyledItem flexDirection="row">
      <StyledImage background={background}>
        <img
          src={imageUrl}
          alt={imgName}
          width={46}
          height={46}
        />
      </StyledImage>
      <Box>
        <StyledText flexDirection="column">
          <Header variant="s" as="h5" mb="xs" text={item.title} linesToShow={2} marginBottom="s" />
          <Flex justifyContent="flex-start" alignItems="center" flexWrap="wrap">
            <Flex justifyContent="flex-start" alignItems="flex-end" width="inherit">
              <EpisodeTags isEpisodeNew={isEpisodeNew} isExplicitContent={item.contentRating === 'Explicit'} />
              {podcastName && <Paragraph variant="m" as="p" mr="xs" transparent text={`${podcastName} - ${formatSecondsToTime(item.durationSeconds)}`} />}
            </Flex>
          </Flex>
        </StyledText>
      </Box>
    </StyledItem>
  );
}

const StyledText = styled(Flex)`
  color: ${props => props.theme.LEGACY_secondary};

  ${screen.md} {
    margin-right: 6px;
  }
`;

const StyledImage = styled(Box)`
  flex-shrink: 0;
  border-radius: 4px;
  margin-right: ${spacing.m};
  overflow: hidden;
  width: 46px;
  height: 46px;
`;

const StyledItem = styled(Flex)`
  width: inherit;
  cursor: pointer;
  text-decoration: none;
  page-break-inside: avoid;
  appearance: none;
  background: transparent;
  border: 0;
  text-align: left;
  padding: ${spacing.m};
`;

TrackItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    imageUrl: PropTypes.string,
    description: PropTypes.string,
    publishedUtc: PropTypes.string,
    contentRating: PropTypes.string,
    durationSeconds: PropTypes.number,
    item: PropTypes.number,
    season: PropTypes.number,
    info: PropTypes.string,
    publishedDate: PropTypes.string,
    playlistDate: PropTypes.string,
    playlistEp: PropTypes.string,
    comment: PropTypes.string,
    commentator: PropTypes.string,
    show: PropTypes.shape({
      images: PropTypes.shape({
        url: PropTypes.string,
      }),
    }),
  }).isRequired,
  background: PropTypes.string,
  imgName: PropTypes.string,
  isEpisodeNew: PropTypes.bool,
  podcastName: PropTypes.string,
};

TrackItem.defaultProps = {
  background: null,
  imgName: '',
  isEpisodeNew: false,
  podcastName: null,
};

export default TrackItem;
