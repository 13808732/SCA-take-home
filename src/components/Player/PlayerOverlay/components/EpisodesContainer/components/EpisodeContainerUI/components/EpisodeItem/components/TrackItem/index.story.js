/* eslint-disable max-len */
import React from 'react';
import { storiesOf } from '@storybook/react';
import PlayIcon from 'components/Icons/play.svg';
import TrackItem from './index';

const smallImgProps = {
  imageUrl: 'https://nowplaying.scadigital.com.au/images/soniadada-soniadada-youdonttreatmenogood.png?station=2day_fm',
  description: ' Jane smuggles kids into a hotel plus Andrew Denton, Mick sings Kiss songs and Rai from Thirsty Merc sings his own songs plus Andrew Denton!',
  subDescription: 'TAG - DD month YYYY - Info',
};

const largeImgProps = {
  imageUrl: 'https://cdn-img.scalabs.com.au/UBGLBNb9nqOSuRx-ijY4AM-JGLE36gdJz5uKgy3LRKs/aHR0cHM6Ly9zdy1o/aXQtcHJkLnNjYWRp/Z2l0YWwuaW8vbWVk/aWEvNjY2MzcvdHJh/aW4tZ2xlbmJyb29r/LXR3aXR0ZXIuanBn/P3ByZXNldD1tcnBy/ZXZhcnRpY2xl',
  description: ' Jane smuggles kids into a hotel plus Andrew Denton, Mick sings Kiss songs and Rai from Thirsty Merc sings his own songs plus Andrew Denton!',
  subDescription: 'TAG - DD month YYYY - Info',
  wideImg: true,
};

storiesOf('Topic Item', module)
  .addWithJSX('Topic item with catchup image', () => (
    <TrackItem {...smallImgProps} />
  ))
  .addWithJSX('Topic item with video or news image', () => (
    <TrackItem {...largeImgProps} />
  ))
  .addWithJSX('Topic item with icon in subdescription', () => (
    <TrackItem {...largeImgProps} icon={PlayIcon} />
  ));
