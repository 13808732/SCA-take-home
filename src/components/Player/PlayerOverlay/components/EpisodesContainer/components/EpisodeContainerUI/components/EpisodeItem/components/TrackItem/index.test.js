/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TrackItem from './index';

describe('<TrackItem /> Component', () => {
  const smallImgProps = {
    item: {
      season: 1,
      episode: 1,
      imageUrl: 'https://nowplaying.scadigital.com.au/images/soniadada-soniadada-youdonttreatmenogood.png?station=2day_fm',
      description: ' Jane smuggles kids into a hotel plus Andrew Denton, Mick sings Kiss songs and Rai from Thirsty Merc sings his own songs plus Andrew Denton!',
      subDescription: 'TAG - DD month YYYY - Info',
    },
    isEpisodeNew: false,
  };

  const largeImgProps = {
    item: {
      season: 2,
      episode: 1,
      imageUrl: 'https://cdn-img.scalabs.com.au/UBGLBNb9nqOSuRx-ijY4AM-JGLE36gdJz5uKgy3LRKs/aHR0cHM6Ly9zdy1o/aXQtcHJkLnNjYWRp/Z2l0YWwuaW8vbWVk/aWEvNjY2MzcvdHJh/aW4tZ2xlbmJyb29r/LXR3aXR0ZXIuanBn/P3ByZXNldD1tcnBy/ZXZhcnRpY2xl',
      description: ' Jane smuggles kids into a hotel plus Andrew Denton, Mick sings Kiss songs and Rai from Thirsty Merc sings his own songs plus Andrew Denton!',
      subDescription: 'TAG - DD month YYYY - Info',
    },
    isEpisodeNew: true,
  };

  test('it should render the Topics for catchup items', () => {
    const component = renderer.create(<TrackItem {...smallImgProps} title={'Lastest catchup'} />);
    const tree = component.toJSON();
    expect(tree)
      .toMatchSnapshot();
  });

  test('it should render the Topics for video or news items', () => {
    const component = renderer.create(<TrackItem {...largeImgProps} title={'Lastest news'} />);
    const tree = component.toJSON();
    expect(tree)
      .toMatchSnapshot();
  });

  test('it should render the Topics with icon for video or news items', () => {
    const component = renderer.create(<TrackItem {...largeImgProps} title="Latest News" icon={<svg />} />);
    const tree = component.toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
