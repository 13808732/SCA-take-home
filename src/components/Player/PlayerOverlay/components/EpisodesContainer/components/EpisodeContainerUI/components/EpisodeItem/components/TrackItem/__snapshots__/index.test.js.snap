// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItem /> Component it should render the Topics for catchup items 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c11 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  font-family: dunbar-tall,sans-serif;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-top: 0px;
  margin-bottom: 3px;
  opacity: 1;
}

.c8 {
  display: block;
  display: -webkit-box;
  max-width: 100%;
  color: inherit;
  padding: 0;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c9 {
  max-height: calc(14px * 1.2 * 2 );
  font-size: 14px;
  line-height: 1.2;
  -webkit-line-clamp: 2;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 4px;
  margin-right: 12px;
  overflow: hidden;
  width: 46px;
  height: 46px;
}

.c2 {
  width: inherit;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  page-break-inside: avoid;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0;
  text-align: left;
  padding: 12px;
}

@media screen and (min-width:40em) {
  .c9 {
    max-height: calc(16px * 1.2 * 2 );
    font-size: 16px;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    margin-right: 6px;
  }
}

<div
  className="c0 c1 c2"
>
  <div
    className="c3 c4"
  >
    <img
      alt=""
      height={46}
      src="https://nowplaying.scadigital.com.au/images/soniadada-soniadada-youdonttreatmenogood.png?station=2day_fm"
      width={46}
    />
  </div>
  <div
    className="c3"
  >
    <div
      className="c5 c1 c6"
    >
      <h5
        className="c7 c8 c9"
      />
      <div
        className="c10 c1"
      >
        <div
          className="c11 c1"
          width="inherit"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<TrackItem /> Component it should render the Topics for video or news items 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c11 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  font-family: dunbar-tall,sans-serif;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-top: 0px;
  margin-bottom: 3px;
  opacity: 1;
}

.c8 {
  display: block;
  display: -webkit-box;
  max-width: 100%;
  color: inherit;
  padding: 0;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c9 {
  max-height: calc(14px * 1.2 * 2 );
  font-size: 14px;
  line-height: 1.2;
  -webkit-line-clamp: 2;
}

.c12 {
  top: .125em;
  position: relative;
}

.c12 svg {
  margin-left: 3px;
  margin-right: 3px;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 4px;
  margin-right: 12px;
  overflow: hidden;
  width: 46px;
  height: 46px;
}

.c2 {
  width: inherit;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  page-break-inside: avoid;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0;
  text-align: left;
  padding: 12px;
}

@media screen and (min-width:40em) {
  .c9 {
    max-height: calc(16px * 1.2 * 2 );
    font-size: 16px;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    margin-right: 6px;
  }
}

<div
  className="c0 c1 c2"
>
  <div
    className="c3 c4"
  >
    <img
      alt=""
      height={46}
      src="https://cdn-img.scalabs.com.au/UBGLBNb9nqOSuRx-ijY4AM-JGLE36gdJz5uKgy3LRKs/aHR0cHM6Ly9zdy1o/aXQtcHJkLnNjYWRp/Z2l0YWwuaW8vbWVk/aWEvNjY2MzcvdHJh/aW4tZ2xlbmJyb29r/LXR3aXR0ZXIuanBn/P3ByZXNldD1tcnBy/ZXZhcnRpY2xl"
      width={46}
    />
  </div>
  <div
    className="c3"
  >
    <div
      className="c5 c1 c6"
    >
      <h5
        className="c7 c8 c9"
      />
      <div
        className="c10 c1"
      >
        <div
          className="c11 c1"
          width="inherit"
        >
          <div
            className="c12"
          >
            <svg
              height="14"
              viewBox="0 0 30 14"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
              >
                <rect
                  height="13"
                  rx="2"
                  stroke="#40A2DB"
                  width="29"
                  x=".5"
                  y=".5"
                />
                <path
                  d="M12.469 7.5c.307-2 .979-3 2.015-3 1.555 0 2.016 2.5 2.016 3"
                  stroke="#40A2DB"
                />
                <path
                  d="M16.5 9.5c-.557.667-1.195 1-1.914 1-1.078 0-2.117-2.04-2.117-3.04"
                  stroke="#40A2DB"
                />
                <path
                  d="M16.5 7.5h-4.031M18.5 4.5l2 6M20.5 10.5l1-5M21.5 5.5l1 5M24.5 4.5l-2 6M4.5 10.5v-7M4.5 3.5l5 7M9.5 10.5v-7"
                  stroke="#40A2DB"
                  strokeLinecap="square"
                />
                <path
                  d="M18 4h1v1h-1zM24 4h1v1h-1z"
                  fill="#40A2DB"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TrackItem /> Component it should render the Topics with icon for video or news items 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c11 {
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  font-family: dunbar-tall,sans-serif;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-top: 0px;
  margin-bottom: 3px;
  opacity: 1;
}

.c8 {
  display: block;
  display: -webkit-box;
  max-width: 100%;
  color: inherit;
  padding: 0;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c9 {
  max-height: calc(14px * 1.2 * 2 );
  font-size: 14px;
  line-height: 1.2;
  -webkit-line-clamp: 2;
}

.c12 {
  top: .125em;
  position: relative;
}

.c12 svg {
  margin-left: 3px;
  margin-right: 3px;
}

.c4 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-radius: 4px;
  margin-right: 12px;
  overflow: hidden;
  width: 46px;
  height: 46px;
}

.c2 {
  width: inherit;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  page-break-inside: avoid;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0;
  text-align: left;
  padding: 12px;
}

@media screen and (min-width:40em) {
  .c9 {
    max-height: calc(16px * 1.2 * 2 );
    font-size: 16px;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    margin-right: 6px;
  }
}

<div
  className="c0 c1 c2"
>
  <div
    className="c3 c4"
  >
    <img
      alt=""
      height={46}
      src="https://cdn-img.scalabs.com.au/UBGLBNb9nqOSuRx-ijY4AM-JGLE36gdJz5uKgy3LRKs/aHR0cHM6Ly9zdy1o/aXQtcHJkLnNjYWRp/Z2l0YWwuaW8vbWVk/aWEvNjY2MzcvdHJh/aW4tZ2xlbmJyb29r/LXR3aXR0ZXIuanBn/P3ByZXNldD1tcnBy/ZXZhcnRpY2xl"
      width={46}
    />
  </div>
  <div
    className="c3"
  >
    <div
      className="c5 c1 c6"
    >
      <h5
        className="c7 c8 c9"
      />
      <div
        className="c10 c1"
      >
        <div
          className="c11 c1"
          width="inherit"
        >
          <div
            className="c12"
          >
            <svg
              height="14"
              viewBox="0 0 30 14"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fillRule="evenodd"
              >
                <rect
                  height="13"
                  rx="2"
                  stroke="#40A2DB"
                  width="29"
                  x=".5"
                  y=".5"
                />
                <path
                  d="M12.469 7.5c.307-2 .979-3 2.015-3 1.555 0 2.016 2.5 2.016 3"
                  stroke="#40A2DB"
                />
                <path
                  d="M16.5 9.5c-.557.667-1.195 1-1.914 1-1.078 0-2.117-2.04-2.117-3.04"
                  stroke="#40A2DB"
                />
                <path
                  d="M16.5 7.5h-4.031M18.5 4.5l2 6M20.5 10.5l1-5M21.5 5.5l1 5M24.5 4.5l-2 6M4.5 10.5v-7M4.5 3.5l5 7M9.5 10.5v-7"
                  stroke="#40A2DB"
                  strokeLinecap="square"
                />
                <path
                  d="M18 4h1v1h-1zM24 4h1v1h-1z"
                  fill="#40A2DB"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
